apps/foundation-demo/src/app/design/atoms/button/button.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit, ViewEncapsulation, ChangeDetectionStrategy, Input, ElementRef } from '@angular/core';
+
+/*
+List of classes to add to Daff Button instances based on host attributes to style as different variants.
+*/
+
+const BUTTON_HOST_ATTRIBUTES = [
+  'daff-button',
+  'daff-stroked-button',
+  'daff-clear-button',
+  'daff-icon-button'
+];
+
+
+@Component({
+  selector: 'button[daff-button], button[daff-stroked-button], button[daff-clear-button], button[daff-icon-button]' + ',' +
+            'a[daff-button], a[daff-stroked-button], a[daff-clear-button], a[daff-icon-button]',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
So you have the styles for a class called `daff-button`, but those styles won't be applied unless you manually add the class:
```
<button daff-button class="**daff-button**">Add to Cart</button>
```
instead, you could add the class to the host button upon creation of a new `daff-button` component. To do this, you add `host: {'class': 'daff-button'}` here.